src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs

```diff
@@ -219,25 +232,27 @@ private long GetOptions(SslProtocols protocols)
                 {
                     outputBuffer.token = new byte[outputBuffer.size];
                     Marshal.Copy(outputPtr, outputBuffer.token, 0, outputBuffer.size);
-                    Marshal.FreeHGlobal(outputPtr);
                 }
                 else
                 {
                     outputBuffer.token = null;
-                }      
+                }
+
+                if (outputPtr != IntPtr.Zero)
+                {
+                     Marshal.FreeHGlobal(outputPtr);
+                     outputPtr = IntPtr.Zero;
+                }
 
                 return done ? SecurityStatus.OK : SecurityStatus.ContinueNeeded;
             }
-            catch
+            catch(Exception Ex)
             {
-                return SecurityStatus.InternalError;
+                Debug.Fail("Exception Caught. StackTrace - " +Ex.StackTrace);
```

Comment:
The other one of these was fixed to print the whole exception (type and message included), not just the stack trace.

Also, `Ex` => `ex`, and whitespace for catch (add a space after the keyword, before the paren) and the + operation (add a space after).